[FEAT] Time Picker 컴포넌트 구현

kamja0510 opened the pull request:
## Related issue 🛠
- closed #28 

## Work Description ✏️
- iOS 스타일의 Wheel Picker 컴포넌트 구현
- Wheel Picker 이용해서 Time Picker 구현

## Screenshot 📸

https://github.com/user-attachments/assets/32c78276-29ac-44b7-a602-3cc07b87977b


## To Reviewers 📢

kamja0510 committed:
[feat] #28 BbangZipTimePicker 구현

kamja0510 committed:
[feat] #28 BbangZipTimePicker Ui 완벽구현

kamja0510 committed:
[feat] #28 BbangZipTimePicker Ui 완벽구현

kamja0510 committed:
[feat] #28 BbangZipWheelPicker 완벽구현

kamja0510 committed:
[refactor] #28 ktlint

kamja0510 committed:
[feat] #28 BbangZipTimePicker.kt 구현 완료

kamja0510 committed:
[refactor] #28 디자인 상수들 추출 완료

kamja0510 committed:
[refactor] #28 localTime 사용으로 변경

kamja0510 committed:
[doc] #28 BbangZipWheelPickerDefaults.kt Kdoc 작성

kamja0510 committed:
[doc] #28 BbangZipTimePicker.kt, BbangZipWheelPicker.kt 주석 작성

kamja0510 committed:
[refactor] #28 derivedState로 현재 시각 관리

kamja0510 committed:
[refactor] #28 ktlint

beom84 commented on app/src/main/java/org/android/bbangzip/presentation/component/timepicker/BbangZipWheelPicker.kt:
```diff
@@ -0,0 +1,181 @@
+        }
+    val scrollOffsetToCenterItem =
+        remember(itemHeightPx, visibleItemsCount) {
+            ((itemHeightPx * visibleItemsCount) / 2f - itemHeightPx / 2f).toInt()
```
p5 : 이런식으로 함수로 빼서 가독성 올리는 방식 어떠신가요?
```suggestion
           private fun calculateCenterOffset(itemHeight: Float, visibleItemsCount: Int): Int {
    val totalHeight = itemHeight * visibleItemsCount
    val centerPosition = totalHeight / 2f
    val itemCenterOffset = itemHeight / 2f
    return (centerPosition - itemCenterOffset).toInt()
}
```

kamja0510 replied:
넵!

kamja0510 committed:
[refactor] #28 BbangZipWheelPicker.kt

calculateCenterOffset 함수화

kamja0510 committed:
[Merge] #28 branch 'develop' into feat/component-time-picker

# Conflicts:
#	app/src/main/java/org/android/bbangzip/presentation/util/extension/LocalTimeExt.kt
#	app/src/main/res/values/strings.xml

kamja0510 committed:
[refactor] #28 ktlint formatting

beom84 reviewed (approved):
어푸어푸
